pkg/activator/net/revision_backends.go

```diff
@@ -189,6 +191,24 @@ func (rw *revisionWatcher) probePodIPs(dests []string) (sets.String, error) {
 	return hs, err
 }
 
+func (rw *revisionWatcher) sendUpdate(update *RevisionDestsUpdate) bool {
+	// In many cases we are doing blocking work while determining whether we have an update.
+	// we could have been asked to shut down and updateCh / doneCh are closed. Force a
+	// check of doneCh first in case this is the situation.
+	select {
```

Comment:
I don't really see how this helps the race condition. It just makes it less probable. But the channels _can_ be closed between end of first select and start of second select. Thus still having potential of sending update on the closed channel.

Reply:
correct. That being said, in practice it seems to completely eliminate this race condition which really only shows up during test cleanup. I don't have a better idea of how to get rid of this but suggestions welcome.

Reply:
Updated per your suggestion, thanks!